internal/storage/v2/elasticsearch/factory_test.go

```diff
@@ -68,3 +72,100 @@ func getTestingFactoryBase(t *testing.T, cfg *escfg.Configuration) *elasticsearc
 	require.NoError(t, err)
 	return f
 }
+
+func newTestConfig(tags escfg.TagsAsFields, serverURL string) escfg.Configuration {
+	return escfg.Configuration{
+		Servers:  []string{serverURL},
+		LogLevel: "error",
+		Tags:     tags,
+	}
+}
+
+// cleanAndSortTags splits a comma-separated tag string, removes empty tags, and sorts the result.
+func cleanAndSortTags(tagString string) []string {
+	tags := strings.Split(tagString, ",")
+	cleanTags := make([]string, 0, len(tags))
+	for _, tag := range tags {
+		if tag != "" {
+			cleanTags = append(cleanTags, tag)
+		}
+	}
+	sort.Strings(cleanTags)
+	return cleanTags
+}
+
+func TestEnsureRequiredFields(t *testing.T) {
```

Comment:
nit: I suggest another test that simply validates (strings.Contains) that once you create a factory with empty TagsAsFields or with TagsAsFields="foo.bar" that the factory will always include the two extra tags. That would be testing the desired outcome, not just the permutations below.